src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/StandardOleMarshalObject.Unix.cs

```diff
@@ -0,0 +1,46 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Runtime.InteropServices
+{
+    public class StandardOleMarshalObject : MarshalByRefObject, IMarshal
+    {
+        protected StandardOleMarshalObject()
+        {
+        }
+
+        int IMarshal.GetUnmarshalClass(ref Guid riid, IntPtr pv, int dwDestContext, IntPtr pvDestContext, int mshlflags, out Guid pCid)
```

Comment:
None of these implementation methods are needed. `IMarshal` is internal interface used just for COM; and COM is not supported on Unix.

This should be just a dummy type with just a constructor.